onnxruntime/core/framework/execution_frame.cc

```diff
@@ -74,6 +76,9 @@ Status IExecutionFrame::GetOrCreateNodeOutputMLValue(int index, const TensorShap
 }
 
 AllocatorPtr IExecutionFrame::GetAllocator(const OrtMemoryInfo& info) const {
+  if (custom_cpu_allocator_ != nullptr && std::strncmp(info.name, onnxruntime::CPU, 3) == 0 && info.mem_type == OrtMemTypeDefault)
+    return custom_cpu_allocator_;
+      
   return GetAllocatorImpl(info);
 }
```

Comment:
Not sure about this setup. Currently there's a strong binding between EPs and allocators. If the need is to provide essentially an override for an allocator, maybe utils::GetAllocator should be providing that ability (and needs to be given the RunOptions in order to do so).